modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition.py

```diff
@@ -109,6 +120,61 @@ def put(cls, obj):
         """
         return cls(obj)
 
+    def insert(
+        self, idx: int, name: str, value: AnyArrayLike
+    ) -> Tuple["HdkOnNativeDataframePartition", pa.lib.DataType]:
+        """
+        Insert column into this raw partition.
+
+        Parameters
+        ----------
+        idx : int
+        name : str
+        value : AnyArrayLike
+
+        Returns
+        -------
+        Tuple[HdkOnNativeDataframePartition, pa.lib.DataType]
+        """
+        data = self._data
+        name = ColNameCodec.encode(name)
+
+        if isinstance(data, pandas.DataFrame):
+            data = data.copy(False)
+            data.insert(idx, name, value)
+            dtype = data.dtypes[idx]
+        elif isinstance(data, pa.Table):
+            try:
+                new_data = data.add_column(idx, name, [value])
+                dtype = new_data.field(idx).type.to_pandas_dtype()
+                data = new_data
+            except Exception:
+                try:
+                    df = pandas.DataFrame({name: value})
+                    at = pa.Table.from_pandas(df, preserve_index=False)
+                    data = data.add_column(idx, at.field(0), at.column(0))
+                    dtype = df.dtypes[0]
+                except Exception as err:
+                    raise NotImplementedError(repr(err))
+        else:
+            raise NotImplementedError(f"Insertion into {type(data)}")
+
+        ensure_supported_dtype(dtype)
+        return HdkOnNativeDataframePartition(data), dtype
```

Comment:
`dtype` should be converted to `pa.lib.DataType` according docstring

Reply:
Fixed the docstring.